src/message/commands/mod.rs

```diff
@@ -318,13 +314,12 @@ impl IRCMessageParseExt for IRCMessage {
         // badges tag format:
         // admin/1,moderator/1,subscriber/12
         for src in tag_value.split(',') {
-            let (name, version) = src
-                .splitn(2, '/')
-                .map(|s| s.to_owned())
-                .next_tuple()
-                .ok_or_else(make_error)?;
+            let (name, version) = src.split_once('/').ok_or_else(make_error)?;
 
-            badges.push(Badge { name, version })
+            badges.push(Badge {
+                name: name.to_string(),
+                version: version.to_string(),
```

Comment:
Just to make it consistent with the rest of the library, I always used `to_owned`

```suggestion
                name: name.to_owned(),
                version: version.to_owned(),
```